src/syrupy/session.py

```diff
@@ -216,5 +219,7 @@ def _count_snapshots(self, snapshot_files: "SnapshotFiles") -> int:
         return sum(len(snaps) for snaps in snapshot_files.values())
 
     @staticmethod
-    def get_node_testname(node: Any) -> str:
-        return str(getattr(node, "name", node.obj.__name__ if node.obj else ""))
+    def snapshot_name_matches_test_node(name: str, node: Any) -> bool:
+        methodname = node.obj.__name__ if node.obj else None
+        nodename = getattr(node, "name", None)
+        return methodname in name or name in nodename
```

Comment:
this will produce false positives for names like: `test_case_1` and `test_case_10`.

Reply:
Not sure to disambiguate this, since `test_case_1` and `test_case_10` can be generated by `test_case_1` with the same serializer

Reply:
Leaving the logic as is, the issue is there's no way to guarantee the serializer still exists, if it does then good we can use it to remove the snapshot. If it does not then we have to still detect that the same function no longer asserts on those snapshots when ran. I think the assumption that method names would match snapshot names to some degree is safe.